pkg/api/helper/helpers.go

```diff
@@ -113,9 +113,18 @@ func IsStandardContainerResourceName(str string) bool {
 }
 
 // IsOpaqueIntResourceName returns true if the resource name has the opaque
-// integer resource prefix.
+// integer resource prefix, or it is another unknown resource name not in the
+// default namespace.
 func IsOpaqueIntResourceName(name api.ResourceName) bool {
-	return strings.HasPrefix(string(name), api.ResourceOpaqueIntPrefix)
+	return IsCustomNamespacedResourceName(name)
```

Comment:
Would this also include "alpha.kubernetes.io/xxx" or ResourceStorageScratch or ResourceStorageOverlay? Generally I feel it is probably safer to start with:
func IsIntResourceName(name api.ResourceName) bool {
    return IsOpaqueIntResourceName(name) || IsExtendedResourceName(name)
}
to make sure we don't break the current resource behavior.